repo-agent/review-ui/ui/src/UpdateRepo.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from 'react';
+import yaml from 'js-yaml';
+
+function UpdateRepo({ repo, onCancel, onRepoUpdated }) {
+    const [yamlContent, setYamlContent] = useState('');
+    const [originalYamlContent, setOriginalYamlContent] = useState('');
+    const [isLoading, setIsLoading] = useState(false);
+    const [error, setError] = useState('');
+
+    useEffect(() => {
+        setIsLoading(true);
+        fetch(`/api/repos/${repo.name}/yaml`)
+            .then(res => {
+                if (!res.ok) throw new Error("Failed to fetch repo YAML");
+                return res.json();
+            })
+            .then(data => {
+                if (data.yaml) {
+                    setYamlContent(data.yaml);
+                    setOriginalYamlContent(data.yaml); // Store original YAML content
+                }
+                setIsLoading(false);
+            })
+            .catch(err => {
+                console.error(err);
+                setError(err.message);
+                setIsLoading(false);
+            });
+    }, [repo.name]);
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        setError('');
+        setIsLoading(true);
+
+        try {
+            // Validate YAML
+            const parsed = yaml.load(yamlContent);
```

Comment:
While this client-side validation to prevent `repoURL` changes is a good UX measure, it should not be relied upon for security. As noted in the backend review, the primary validation must happen on the server. This comment could be updated to reflect that it's a UX helper, not a security control.